proposed/2023/release-staging-and-deprecation.md

```diff
@@ -0,0 +1,199 @@
+ have changed after staging, should result in an error with details about the change. The `--force` flag should again be used
+ to ignore the change and move forward with execution of the requst (either publishing or deprecation). To summarize,
+ the fundamental principle is that a user cannot deprecate a package they don't own.
+
+#### Lifespan of a staged group
+If users keep staging packages but do nothing with them, NuGet will become an ever-growing package graveyard,
+ incurring higher and higher costs over time simply for storing these packages. To avoid this problem, staging should be time-limited.
+ That is, there should be a maximum lifespan for a staged package, that is not published. NuGet should then unstage / remove
+ any package that was staged but never published within the specified / pre-configured time period. As a maximum timeslot
+ a default one month period can be considered. However, to reduce the cost of maintaining these packages,
+ the `nuget stage` command can also accept an optional `--lifespan [number of days]` parameter,
+ which can indicate a shorter time period, thus giving NuGet ability to remove those packages earlier than the maximum period.
+
+Note, that this opens up a new scenario, where a group has been created with a package 29 days ago, and then a new package is staged to the same group.
+In two days from that time, the package author tries to publish the staged group. At this point, the publish command will fail,
+indicating that the group has been modified, and showing the package that has been deleted by NuGet out of the staging due to the expiration.
+The user can still continue publishing using the `--force` option, which will result in the non-expired/ removed staged packages to be published.
+Alternatively, the user can restage the expired package and call the publish command again.
+
+#### Expanding groups
+It should be possible for a group to expand over time. This is currently a real scenario, where .NET GA SDK ships a set of packages,
+ and then follow-up SDK updates ship updated packages which still belong to the same .NET 8 package group.
+ Here how this will play out with the tooling support described above:
+1. During GA release, the release team will stage the set of RTM packages for the release. (nuget stage --group-id "net8.0" --package-id <package-id>)
+2. Then on the day of the release, all the staged packages will be published using `nuget publish --group-id "net8.0"` command.
+   At this point, the stage for the `net8.0` group will be empty.
+3. Later, as new builds are being prepared for a patch release, a new set of packages will be staged, to be later published to the same group.
+As there can be multiple candidate builds for a release, there is a need for functionality to discard a stage for a given group `nuget stage --discard --group-id "net8.0"`.
```

Comment:
I think it's a little inconsistent to have the `discard` verb hidden as an option `--discard`. It almost feels like we want `dotnet nuget stage discard` or (perhaps more aligned with .NET CLI standards) `dotnet nuget remove stage`. I'm not an expert here but it seems inconsistent with current .NET CLI or even `dotnet nuget --help` patterns.

Reply:
Updated to us nuget delete command option instead.